### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by FenrisStargazer**
Pull request for the sake of meeting the deadline even though it means next to nothing... There's nothing here. I'm utterly lost and confused and feel like a complete idiot. Hopefully I can figure it out before the last deadline...

Website link: `abrand3-wordle.azurewebsites.net` (Even though it doesn't work...)

### CODE REVIEW

**Comment by VForev** [comment]
Your not alone

**Comment by NicholasBurlakov** [comment]
To be honest i too was super lost and felt hopeless. But rewatching the videos about the azure deployment and SQL manipulation helped a ton. Don't call yourself an idiot, Im sure half the class is just as lost. Ask Garret questions, he is super kind and helpful when it comes to this stuff. I hope you figure it out and wish you all the best!

File: .github/workflows/assignment-3_abrand3-wordle.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# More GitHub Actions for Azure: https://github.com/Azure/actions
```
Comment (j0shh3ss) [inline .github/workflows/assignment-3_abrand3-wordle.yml:2]: This assignment isn't easy! Don't be hard on yourself, you can do this!

**Review by j0shh3ss — COMMENTED** [review]
1. Create a leaderboard API Controller ❌

Create an endpoint that returns the top 10 scores (HttpGet) ❌
Results should include Name, average guesses, and number of games played ❌
Create an endpoint that allows for posting a score (HttpPost) with data in the body ❌
A new name should add a new record ❌
An existing name should update that record ❌
Save and load all data with Entity Framework to an Azure SQL database or LocalDb for debugging ❌
Class should be named "Player" and "Players" for the DbSet ❌
Fields should be:
int PlayerId ❌
string Name ❌
int GameCount ❌
double AverageAttempts ❌
int AverageSecondsPerGame ❌(OPTIONAL)
The logic should reside in a service that is injected via dependency injection into the controller ❌
Set up CORS (Cross Origin Resource Sharing) ❌
2. Create a dialog for getting the user's name ❌

If we don't have the user's name, present the user with a dialog that allows them to type in their name ❌
The users current name should show on the game page (upper right) ❌
This should be visible when setting/changing the name. It should change as the user types ❌
The user should be able to click on their name to change it (with the same dialog) ❌
Save the name in local storage and load it for the next game if it is set ❌
If the name is not set, display Guest as the name. ❌
Clicking on this allows for preemptive setting of the name before the end of the game ❌
Dialog must be persistent, doesn't close by clicking off the dialog ❌
3. Save the Score and Seconds to complete ❌

Save the score and time to complete in seconds at the end of the game if we have a name. ❌
If the user hasn't set a name, save the score and seconds as "Guest". (Be sure to prompt first) ❌
Save should be done with an Axios Post ❌
4. Create a Leaderboard Page ❌

Create a /leaderboard page in the application that shows the top scores ❌
Use Axios to call the leaderboard API ❌
Make a nice display of the returned data ❌
Add a link on the front page to access the leaderboard ❌
Add a link to the app bar to access the leaderboard ❌
5. Unit testing ❌ (OPTIONAL) (Not covered in class, but in instructor's repo: https://github.com/granterickson/EWU-CSCD379-2023-Spring)

Tests for the service ❌ (OPTIONAL)
Integration tests for the controller ❌ (OPTIONAL)
6. Deploy site to Azure ❌

Deploy API site with Github Action to Azure Web App ❌
Unit Tests should run on build ✅
Add API Base URL replacement to front end project via secret ✅

**Comment by FenrisStargazer** [comment]
Azure stuff doesn't wanna work but if you run it locally it's almost kinda not the worst thing ever. Barely.

File: Wordle.Api/Wordle.Api/Controllers/PersonController.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+using System.Numerics;
+using Wordle.Api.Data;
+using Wordle.Api.Dtos;
+using Wordle.Api.Services;
+
+namespace Wordle.Api.Controllers;
+
+[ApiController]
+[Route("Player")]
+public class PlayerController : ControllerBase
```
Comment (mmwoodfo) [inline Wordle.Api/Wordle.Api/Controllers/PersonController.cs:12]: Your file name is `PersonController.cs` but your class name is `PlayerController`, this is causing conflicts with your other file, called `PlayerController.cs`

image

**Review by mmwoodfo — COMMENTED** [review]
After fixing your code so I could launch the API locally, it looks good. I think you got it all working!  Not sure what the azure issue is, but it looks like you may need to generate a new deployment token? 

image


You can watch the lecture video from the lecture about setting up the Azure service, I think this documentation may also explain how to do it: 
https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/static-web-apps/deployment-token-management.md

**Pull request closed without merging**